Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialSystemComponent.cpp

```diff
@@ -135,11 +135,10 @@ namespace AZ
             m_materialPreviewModelAsset.Release();
             m_materialPreviewLightingPresetAsset.Release();
 
-            if (m_openMaterialEditorAction)
-            {
-                delete m_openMaterialEditorAction;
-                m_openMaterialEditorAction = nullptr;
-            }
+            delete m_openMaterialEditorAction;
+            m_openMaterialEditorAction = nullptr;
+            delete m_openMaterialCanvasAction;
+            m_openMaterialCanvasAction = nullptr;
```

Comment:
nit: this code construct indicates that a unique_ptr might help ensure nothing leaks.

Reply:
Updated